src/Daemon/Daemon.cpp

```diff
@@ -119,16 +119,53 @@ namespace usbguard
     _config.close();
   }
 
-  void Daemon::loadConfiguration(const std::string& path)
+  int Daemon::checkPermissions(const std::string& path,
+                               const mode_t permissions)
+  {
+    struct stat file_stat;
+    mode_t permission_bad { ( S_IRWXU | S_IRWXG | S_IRWXO ) - permissions };
+
+    if(!stat(path.c_str(), &file_stat)) {
+      if (S_ISREG(file_stat.st_mode)) {
+        if (  !( file_stat.st_mode & permissions ) ||
+               ( file_stat.st_mode & permission_bad)
+           )
+        {
+          USBGUARD_LOG(Error) << "Policy leaks are possible with this permissions!";
+          throw Exception("Check permissions", path, "Policy leaks possible");
+        }
+        else {
+          USBGUARD_LOG(Info) << "File has correct permissions.";
+        }
+      }
+      else {
+        USBGUARD_LOG(Error) << "ERROR: In specified path is not located file.";
```

Comment:
I don't understand the error message. S_ISREG == false doesn't mean that there's not file at the location, it means that there is one but it's not a regular file. Please provide better wording based on the check that you do on the file.